### ISSUE TITLE
Reference of type 'Microsoft.CodeAnalysis.UnresolvedMetadataReference' is not valid for this compilation when generating metrics.

### ISSUE THREAD

**Issue by fforjan** [dotnet/roslyn#36837]
#### Analyzer package
Microsoft.CodeAnalysis.Metrics

#### Package Version
v2.9.1

#### Repro steps
1. Run the program in command line :
"C:\Users\ISYS36005\TFSWorkspaces\Dev\Source\packages\SchneiderElectric.Software.CodeMetrics.2.0.0\Metrics\metrics.exe" /project:"C:\Users\ISYS36005\TFSWorkspaces\Dev\Source\Ourproject\OurProject.csproj" /OUT:"C:\Users\ISYS36005\TFSWorkspaces\Dev\Source\Ourproject\bin\Debug\OurProject.dll.CodeMetrics.xml"

#### Expected behavior
it should generate the code metrics and if failing, give more details about what is wrong !

#### Actual behavior
Loading OurProject.csproj...
Computing code metrics for OurProject.csproj...
Reference of type 'Microsoft.CodeAnalysis.UnresolvedMetadataReference' is not valid for this compilation.
  Parameter name: references[0]

**Comment by jasonmalinowski** [comment]
@mavasani So what's the bug in the Roslyn context: MSBuildWorkspace is blowing up and it shouldn't be?

**Comment by mavasani** [comment]
Yes, the solution has one C++/CLI project, and MSBuild workspace is blowing up
